api/collections/permissions.py

```diff
@@ -2,11 +2,29 @@
 from __future__ import unicode_literals
 
 from rest_framework import permissions
-from rest_framework.exceptions import NotFound
+from rest_framework.exceptions import NotFound, MethodNotAllowed
 
 from api.base.utils import get_user_auth, assert_resource_type
 from osf.models import AbstractNode, Preprint, Collection, CollectionSubmission, CollectionProvider
-from osf.utils.permissions import WRITE, ADMIN
+from osf.utils.permissions import READ, WRITE, ADMIN
+
+
+class CollectionReadOrPublic(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if request.method == 'GET':
+            return self.has_object_permission(request, view, view.get_object())
+        else:
+            raise MethodNotAllowed(request.method)
+
+    def has_object_permission(self, request, view, obj):
+        auth = get_user_auth(request)
+        moderators = obj.target.collection.provider.get_group('moderator').user_set.all()
+        if auth.user in moderators or obj.target.collection.is_public and obj.target.guid.referent.can_view(auth):
```

Comment:
let's use [`user.has_perms('view_submissions', provider)`](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/registrations.py#L427) instead of loading the whole moderator set? Right now, Provider Admins wouldn't be able to view these.